pkg/controllers/unsupported/unsupported_controller_test.go

```diff
@@ -0,0 +1,24 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package unsupported
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+)
+
+var _ = Describe("CAPI unsupported controller", func() {
```

Comment:
No tests?

Reply:
I wanted to make sure the approach was correct before starting to write tests. If we are happy with the controller, I can write some 👍

Reply:
Tests added